runtime/gc_base/ObjectAccessBarrier.cpp

```diff
@@ -1413,17 +1426,24 @@ MM_ObjectAccessBarrier::structuralCompareFlattenedObjects(J9VMThread *vmThread,
 			}
 
 		} else {
-			if (*(fj9object_t *)((UDATA)lhsObject + startOffset + offset) != *(fj9object_t *)((UDATA)rhsObject + startOffset + offset)) {
-				result = false;
-				break;
+			if (sizeof(uint32_t) == referenceSize) {
+				if (*(uint32_t *)((UDATA)lhsObject + startOffset + offset) != *(uint32_t *)((UDATA)rhsObject + startOffset + offset)) {
```

Comment:
For better readablily, I'd store '(UDATA)lhsObject + startOffset + offset' into a local like lhsAddress.

Reply:
Done.